packages/plugin-client-drizzle/test/integration.ava.ts

```diff
@@ -0,0 +1,2187 @@
+
+  const users = await db
+    .select({
+      name: sql<string>`upper(${usersTable.name})`
+    })
+    .from(usersTable);
+
+  t.deepEqual(users, [{ name: 'JOHN' }]);
+});
+
+test.skip('select distinct', async (t) => {
+  const { db } = t.context;
+
+  const usersDistinctTable = pgTable('users_distinct', {
+    id: integer('id').notNull(),
+    name: text('name').notNull()
+  });
+
+  await db.execute(sql`drop table if exists ${usersDistinctTable}`);
+  await db.execute(sql`create table ${usersDistinctTable} (id integer, name text)`);
+
+  await db.insert(usersDistinctTable).values([
+    { id: 1, name: 'John' },
+    { id: 1, name: 'John' },
+    { id: 2, name: 'John' },
+    { id: 1, name: 'Jane' }
+  ]);
+  const users1 = await db
+    .selectDistinct()
+    .from(usersDistinctTable)
+    .orderBy(usersDistinctTable.id, usersDistinctTable.name);
+  const users2 = await db
+    .selectDistinctOn([usersDistinctTable.id])
+    .from(usersDistinctTable)
+    .orderBy(usersDistinctTable.id);
+  const users3 = await db
+    .selectDistinctOn([usersDistinctTable.name], { name: usersDistinctTable.name })
+    .from(usersDistinctTable)
+    .orderBy(usersDistinctTable.name);
+
+  await db.execute(sql`drop table ${usersDistinctTable}`);
+
+  t.deepEqual(users1, [
+    { id: 1, name: 'Jane' },
+    { id: 1, name: 'John' },
+    { id: 2, name: 'John' }
+  ]);
+
+  t.deepEqual(users2.length, 2);
+  t.deepEqual(users2[0]?.id, 1);
+  t.deepEqual(users2[1]?.id, 2);
+
+  t.deepEqual(users3.length, 2);
+  t.deepEqual(users3[0]?.name, 'Jane');
+  t.deepEqual(users3[1]?.name, 'John');
+});
+
+test.skip('insert returning sql', async (t) => {
+  const { db } = t.context;
+
+  const users = await db
+    .insert(usersTable)
+    .values({ name: 'John' })
+    .returning({
+      name: sql`upper(${usersTable.name})`
+    });
+
+  t.deepEqual(users, [{ name: 'JOHN' }]);
+});
+
+test.skip('delete returning sql', async (t) => {
+  const { db } = t.context;
+
+  await db.insert(usersTable).values({ name: 'John' });
+  const users = await db
+    .delete(usersTable)
+    .where(eq(usersTable.name, 'John'))
+    .returning({
+      name: sql`upper(${usersTable.name})`
+    });
+
+  t.deepEqual(users, [{ name: 'JOHN' }]);
+});
+
+test.skip('update returning sql', async (t) => {
+  const { db } = t.context;
+
+  await db.insert(usersTable).values({ name: 'John' });
+  const users = await db
+    .update(usersTable)
+    .set({ name: 'Jane' })
+    .where(eq(usersTable.name, 'John'))
+    .returning({
+      name: sql`upper(${usersTable.name})`
+    });
+
+  t.deepEqual(users, [{ name: 'JANE' }]);
+});
+
+test.skip('update with returning all fields', async (t) => {
+  const { db } = t.context;
+
+  const now = Date.now();
+
+  await db.insert(usersTable).values({ name: 'John' });
+  const users = await db.update(usersTable).set({ name: 'Jane' }).where(eq(usersTable.name, 'John')).returning();
+
+  t.assert(users[0]!.createdAt instanceof Date);
```

Comment:
Xata contains the right timestamp. However, the timestamp format differs from the one we have for xata internal fields. For example, created_at is 2023-10-06T09:53:49.276Z but our internal xata.creadtedAt field contains 2023-10-06T09:53:49.276755Z (note the precision difference). Mayb that's why datetime parsing fails and we get invalid date errors?